apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
 
-The graph module describes how **entities** can be created, queried, updated, and linked together – including the **block entity**, represented by the **properties** sent to the block.
+The graph module describes how **ontology types** and **entities** can be created, queried, updated, and linked together – including the **block entity**, represented by the **properties** sent to the block.
 
 ## Glossary
 
-**Block entity**: an instance of data associated with the block, conforming to the block’s schema.
+**Entity:** an instance of data, often corresponding to a thing in the world, constrained by a particular entity type.
 
-**Entity:** an instance of data, often corresponding to a thing in the world, of a particular entity type.
+**Link Entity:** a kind of entity which represents a connection or relation between its left and right entity, constrained by a link entity type.
 
-**Entity type:** a specific type of entity, with a schema defining its structure. Entities have a type, and types have a schema.
+**Left Entity:** the entity which is on the "left" of a specific link entity, in most situations this can be thought of as the "source" of the link.
 
-**Graph:** a network of entities connected by links.
+**Right Entity:** the entity which is on the "right" of a specific link entity, in most situations this can be thought of as the "target" of the link.
 
-**Link:** a connection or relation between two entities.
+**Ontology Type:** an element of the Block Protocol Type System, represented as a self-contained schema accessible via a URI. Ontology types are composed together to describe the shape of entities within the graph.
 
-**Linked entities**: entities which are linked to the block entity, whether directly or via other entities.
+**Entity type:** an ontology type that defines the structure of an entity and its link entities. An entity type references other entity types and property types.
 
-**Properties:** the fields on a schema, what types of data are permitted as their values, and any other validation rules. Field types may be scalar (integer, string etc.), complex objects, collections (lists/arrays), or refer to another schema.
+**Link Entity type:** an entity type which has a special marker to say it constrains a link entity.
 
-**Schema:** a definition of a data structure, associated with an entity type. Entities of a given type must have properties which conform to the type’s schema.
+**Property type:** an ontology type that defines the structure of an individual property within an entity. A property type references other property types and data types.
+
+**Data type:** an ontology type that defines the possible space of possible values that an individual property can take within an entity.
+
+**Graph:** a network of graph elements (entities, entity types, property types, and data types) connected by edges.
+
```

Comment:
Vertex? Root?

Reply:
I was trying to keep the glossary from exploding, but perhaps that's not a good idea. I'm also not entirely sure about the glossary, vs the definitions in the spec below, vs the docs elsewhere.

Happy to add sections if you think it necessary, the `Vertex` bit should mostly be an implementation detail, and the `roots` are explained in the context of the `Subgraph` two paragraphs down.

Reply:
It's fine without